### ISSUE TITLE
Irresponsible Module doesn't work when only method calls present

### ISSUE THREAD

**Issue by AndrewRayCode** [troessner/reek#1227]
```
$ reek --version

    reek 4.6.1

$ cat file.rb

    class Thing
    end
    
$ reek file.rb

    file.rb -- 1 warning:
        [1]:IrresponsibleModule: Thing has no descriptive comment [https://github.com/troessner/reek/blob/master/docs/Irresponsible-Module.md]

$ cat otherfile.rb

    class Thing
      does :some_thing
    end
    
$ reek otherfile.rb

    Inspecting 1 file(s):
    .
```

**Comment by troessner** [comment]
Wow wow wiwa! That is definitely a problem. I just reproduced this with the following spec that I appended to our existing spec for IrresponsibleModule:

```Ruby
  it 'reports undocumented classes with method calls in it' do
      src = <<-EOS
        class Alfa
          does :some_thing
        end
      EOS

      expect(src).to reek_of(:IrresponsibleModule)
  end
```

I did check if we didn't have an existing spec that would cover this case and indeed we have one that is at least a bit similar, which you can find [here](https://github.com/troessner/reek/blob/master/spec/reek/smell_detectors/irresponsible_module_spec.rb#L175). Looking at the spec:

```Ruby
    it 'does not report constants that are not classes' do
      src = <<-EOS
        #{scope} Alfa
          Bravo = 23
          Charlie = Hash.new
          Delta = ''.freeze
          Echo = Class.new.new
        end
      EOS

      expect(src).not_to reek_of(:IrresponsibleModule)
    end
```

I don't understand why we had this one in the first place. The class in the spec is not just a namespace module so it should have been reported. @mvz can you recall why we configured IrresponsibleModule like this? My first impression is that this looks definitely like a bug that we should fix.

**Comment by mvz** [comment]
It seems the intent of that spec was that Bravo, Charlie, Delta and Echo would not be reported. Something clearly went wrong with Alfa, though.

**Comment by AndrewRayCode** [comment]
It was introduced here https://github.com/troessner/reek/pull/595/files It seems like two of the added specs here are wrong, because two un-commented tests starting with `module Qux` don't reek

**Comment by mvz** [comment]
Yes, it took me a while to understand my line of reasoning from back then, but it's the namespace module thing that's tripping us up here. The specs are not very clear about that part, because it is incidental to what's being tested right there.

I propose the following two fixes:

* Update the specs to make it clearer when there's a namespace module.
* Update the check for namespace modules to *only* allow constant assignment and class/module definitions. Right now, it allows everything *except* method definitions.

**Comment by troessner** [comment]
@mvz +1 on your proposal. Are you gonna tackle this?

**Comment by mvz** [comment]
Yes, I am.

**Comment by mvz** [comment]
This is fixed in Reek 4.7.1 🎉
